Tests/EngagedTimeTests.swift

```diff
@@ -68,6 +68,9 @@ class EngagedTimeTests: ParselyTestCase {
     
    
     func testGlobalPause() {
+        // This is call to configure required for the start-stop mechanism to work
+        sharedInstance?.configure(siteId: ParselyTestCase.testApikey)
```

Comment:
What do you think changing this test case to use `ParselyTestCase.parselyTestTracker` instead of the singleton? You'll still need to call this configure function, but at least there is no potential side effect to other test cases.

Reply:
Good point. I used `sharedInstance` because that's the object the tests run assertions again, but that doesn't mean we cannot rewrite the test to use the test-scoped value.

I got carried away and did a bunch of refactoring around it in https://github.com/Parsely/AnalyticsSDK-iOS/pull/78, which is on top of this PR.